cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/nosql/NoSqlStructuredTable.java

```diff
@@ -126,6 +131,84 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit) throws I
                              limit);
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> orderByField, SortOrder sortOrder)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: Scan range {} with limit {}, sort field {} and sort order {}", schema.getTableId(),
+              keyRange, limit, orderByField, sortOrder);
+    fieldValidator.validateField(orderByField);
+    if (!schema.isIndexColumn(orderByField.getName())) {
```

Comment:
based on our other discussion, should we also change this to allow it to be one of the primary key fields as well? Or will that be in a different PR?

Reply:
So @de-lan and I discussed this offline, [looks like ](https://github.com/cdapio/cdap/blob/develop/cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTable.java#L135) does this already (sorts on primary keys)

Reply:
Yes, I think the scan methods return sorted results already